src/OpenTelemetry/Trace/Export/Internal/MultiProcessor.cs

```diff
@@ -0,0 +1,60 @@
+﻿// <copyright file="MultiProcessor.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OpenTelemetry.Trace.Export
+{
+    internal class MultiProcessor : SpanProcessor
+    {
+        private readonly IEnumerable<SpanProcessor> processors;
+
+        public MultiProcessor(IEnumerable<SpanProcessor> processors)
+        {
+            this.processors = processors ?? throw new ArgumentNullException(nameof(processors));
+        }
+
+        public override void OnEnd(Span span)
+        {
+            foreach (var processor in this.processors)
+            {
+                processor.OnEnd(span);
+            }
+        }
+
+        public override void OnStart(Span span)
+        {
+            foreach (var processor in this.processors)
+            {
+                processor.OnStart(span);
+            }
+        }
+
+        public override Task ShutdownAsync(CancellationToken cancellationToken)
+        {
+            var tasks = new List<Task>();
+            foreach (var processor in this.processors)
+            {
+                tasks.Add(processor.ShutdownAsync(cancellationToken));
```

Comment:
perhaps some protection from processors returning `null` task can be added here.

Reply:
so "WhenAll" will not crash

Reply:
still, protecting from anything in multi-processor makes no sense (it's optional and only present when advanced config with multiple processors is used), if some processor is misbehaving - we should not hide it here. may be catch in `span.End()`

Reply:
i did exception handling in start/stop, but not her. We don't await WhenAny here and chances are it throws OperationCancelledException, so users will have to try/catch it.